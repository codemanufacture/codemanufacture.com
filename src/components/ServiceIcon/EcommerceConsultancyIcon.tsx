import * as React from 'react'
import * as Icon from './images/ecommerce-consultancy.svg'
import ServiceIcon from './ServiceIcon'

const EcommerceConsultancyIcon = () => (
  <ServiceIcon icon={Icon} name="Ecommerce consultancy" />
)

export default EcommerceConsultancyIcon
