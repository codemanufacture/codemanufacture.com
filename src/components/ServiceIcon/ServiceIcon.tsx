import * as React from 'react'

export interface ServiceIconProps {
  icon: string
  name: string
}

const ServiceIcon: React.FunctionComponent<ServiceIconProps> = ({
  icon,
  name,
}) => (
  <li>
    <figure>
      <img src={icon} alt={`${name}`} />
      <figcaption>{name}</figcaption>
    </figure>
  </li>
)

export default ServiceIcon
