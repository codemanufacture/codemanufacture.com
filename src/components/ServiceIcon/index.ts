export { default as MagentoBuildIcon } from './MagentoBuildIcon'
export { default as ResponsiveDesignIcon } from './ResponsiveDesignIcon'
export { default as EcommerceConsultancyIcon } from './EcommerceConsultancyIcon'
export { default as PowerfulExtensionsIcon } from './PowerfulExtensionsIcon'
export { default as ExternalIntegrationsIcon } from './ExternalIntegrationsIcon'
export { default as MagentoSupportIcon } from './MagentoSupportIcon'
