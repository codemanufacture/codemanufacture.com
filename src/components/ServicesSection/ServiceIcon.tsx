import * as React from 'react'
import { ServicesElement } from '../../enums/services'

interface ServiceIconInterface {
  service: ServicesElement
}

const ServiceIcon: React.SFC<ServiceIconInterface> = ({ service }) => (
  <li>
    <figure>
      <img src={`/assets/${service.iconName}.svg`} alt={service.name} />
      <figcaption>{service.name}</figcaption>
    </figure>
  </li>
)

export default ServiceIcon
