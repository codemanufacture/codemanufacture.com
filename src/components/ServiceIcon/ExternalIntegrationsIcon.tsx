import * as React from 'react'
import * as Icon from './images/external-integrations.svg'
import ServiceIcon from './ServiceIcon'

const ExternalIntegrationsIcon = () => (
  <ServiceIcon icon={Icon} name="External integrations" />
)

export default ExternalIntegrationsIcon
