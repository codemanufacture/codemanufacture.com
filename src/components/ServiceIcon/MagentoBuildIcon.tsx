import * as React from 'react'
import * as Icon from './images/magento-build-and-design.svg'
import ServiceIcon from './ServiceIcon'

const MagentoBuildIcon = () => (
  <ServiceIcon icon={Icon} name="Magento build & design" />
)

export default MagentoBuildIcon
