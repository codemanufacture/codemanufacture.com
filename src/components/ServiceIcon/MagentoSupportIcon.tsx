import * as React from 'react'
import * as Icon from './images/magento-support.svg'
import ServiceIcon from './ServiceIcon'

const MagentoSupportIcon = () => (
  <ServiceIcon icon={Icon} name="Magento support" />
)

export default MagentoSupportIcon
