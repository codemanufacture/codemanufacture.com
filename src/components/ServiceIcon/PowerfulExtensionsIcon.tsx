import * as React from 'react'
import * as Icon from './images/powerful-extensions.svg'
import ServiceIcon from './ServiceIcon'

const PowerfulExtensionsIcon = () => (
  <ServiceIcon icon={Icon} name="Powerful extensions" />
)

export default PowerfulExtensionsIcon
