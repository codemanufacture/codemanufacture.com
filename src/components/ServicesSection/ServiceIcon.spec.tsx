import * as React from 'react'
import renderer from 'react-test-renderer'
import ServiceIcon from './ServiceIcon'

describe('ServiceIcon component', () =>
  it('renders correctly', () => {
    const props = {
      service: {
        iconName: 'icon',
        name: 'service',
      },
    }

    const tree = renderer.create(<ServiceIcon {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
