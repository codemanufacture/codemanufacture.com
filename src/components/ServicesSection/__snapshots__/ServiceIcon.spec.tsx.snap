// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceIcon component renders correctly 1`] = `
<li>
  <figure>
    <img
      alt="service"
      src="/assets/icon.svg"
    />
    <figcaption>
      service
    </figcaption>
  </figure>
</li>
`;
